libpod/container.go

```diff
@@ -706,6 +710,35 @@ func (c *Container) Routes() ([]types.Route, error) {
 	return routes, nil
 }
 
+// BindMounts retrieves bind mounts that were created by libpod and will be
+// added to the container
+// All these mounts except /dev/shm are ignored if a mount in the given spec has
+// the same destination
+// These mounts include /etc/resolv.conf, /etc/hosts, and /etc/hostname
+// The return is formatted as a map from destination (mountpoint in the
+// container) to source (path of the file that will be mounted into the
+// container)
+// If the container has not been started yet, an empty map will be returned, as
+// the files in question are only created when the container is started.
+func (c *Container) BindMounts() (map[string]string, error) {
+	if !c.locked {
+		c.lock.Lock()
+		defer c.lock.Unlock()
+
+		if err := c.syncContainer(); err != nil {
+			return nil, err
+		}
+	}
+
+	newMap := make(map[string]string, len(c.state.BindMounts))
```

Comment:
Should we be adding/creating bind mounts if we can't call syncContainer() due to it being locked by someone else?  I'm just wondering if lines 735 -> 737 should be in the bottom of the above if statement for the lock and then move the map declaration above the if statement.

Reply:
c.locked is used to indicate that the lock is already held as we're being called as part of a batch operation, so this should never happen unlocked.

Reply:
I concur about this not happening unlocked and I'm probably poking at a corner case.  If the container is not locked already you call syncContainer() and it could fail and then the mount doesn't happen.  However if the container is already locked, this code doesn't check the container via syncContainer() to see if it would fail before doing the mount.  Is it a safe assumption that syncContainer would never fail once the container is locked in such a way that adding a bind mount would be inappropriate?

Again most likely a corner case.

Reply:
This isn't actually mounting, it's just getting a list of things we will mount into the container when it is started. Actual mounting is handled by runc.

Reply:
OK yes, but the point still is would adding the list to the container at a point when syncContainer() might have failed if called problematic?  It sounds like you're not concerned, and if that's the case I'm fine with it as is.   But then I wonder if the call to syncContainer() is necessary when the container is unlocked when this routine is called at all?  Something to ponder, your call.

Reply:
This is purely an accessor, and shouldn't be modifying the container at all. It syncs if the container is not in a Batch() to pick up updates from the DB (not necessary in Batch() because we are locked, and no one could have changed the container but us), copies the map of bind mounts (so the user never gets a pointer to our copy and can't change it), and returns the copy